### ISSUE TITLE
Transforms for Connection Strings ignored by XSD queries

### ISSUE THREAD

**Issue by dlee1964** [Microsoft/slow-cheetah#68]
Using VS 2015, Application in test still uses the connection string for debug even though the previewed transform is showing the proper connection string for the test environment.  After moving the queries out of the xsd designer and doing them in code, the application connects to the proper database.  Essentially, XSD based queries ignore the slowcheetah transform.

**Comment by jviau** [comment]
Is this a web application? When debugging C# web projects, the web.config from your source is used directly and *not* the transformed web.config SlowCheetah puts in the output directly.

**Comment by dlee1964** [comment]
No, this is not a web application.

**Comment by dlee1964** [comment]
I don't get why you closed the other issue.  Is it by design that ApplicationSettings Transforms are ignored also in a Windows Forms Application?

**Comment by jviau** [comment]
@dlee1964  I believe this issue and #69 are the same. I would like to point out slow-cheetah is not the correct place for XSD issues. Slow-cheetah uses XDT, but does not own it. You can find XDT help [here](https://msdn.microsoft.com/en-us/library/dd465326(v=vs.110).aspx). That is not to say this isn't a slow-cheetah issue though, it could be the transform is not being applied at all by slow-cheetah.

Can you verify if your file is being transformed? Do _some_ parts of the transform get applied? If not then it isn't specific parts of your transform is ignore, the whole thing is!

If you can verify the transform is being applied, then most likely your transform file is incorrect. In which case I recommend reading the help page I linked earlier. One idea off the top of my head is maybe an xml namespace issue in your transform file? If the elements you are transforming come from a namespace you need to fully qualify it in the transform file.

**Comment by dlee1964** [comment]
No, Issue [#69](https://github.com/Microsoft/slow-cheetah/issues/69) is not the same at all.  Not even close.  Please re-read it and re-open it.

Yes, the preview showed that the transforms were correct.  I was getting no error on the transform.  The same connection string was to be used for XSD and non-XSD based queries.  They were each hitting different databases, yet there was only one connection string in the app.config.  So, I know the transform was correct because the non-XSD queries were hitting the correct database while the XSD based queries weren't.

**Comment by jviau** [comment]
@dlee1964 as far as it concerns slow-cheetah, these are the same issue. The differences between them are in areas slow-cheetah is not responsible for. The issue is with the project system you are using, with XSD itself, or with the way you wrote the transform. Slow-cheetah is not responsible for any of those issues.

First, was the resulting file in your output directory correct after a build? Was the transform applied to it? If so, then the issue is with the project system is my guess. Maybe it does not read the file from the output directory during testing, it could be using the un-transformed file directly from your source tree.

**Comment by dlee1964** [comment]
As I have been in active development and up against a deadline, I had to work around the issues and thus change things significantly, so I cannot answer the questions with confidence at the moment.   I will have to create a small test project to duplicate the issues again to answer your questions.  As I have already stated that the coded queries were using the correct transform and the XSD queries were not (within the same project), I suspect the output file was correct and the XSD was ignoring it.  In any case I will create the small project to demonstrate both issues and perhaps I could upload it for you to review at some point.
